lightning/cmd/tidb-lightning/main.go

```diff
@@ -99,7 +99,9 @@ func main() {
 	finished := true
 	if common.IsContextCanceledError(err) {
 		err = nil
-		finished = false
+		if app.TaskCanceled() {
```

Comment:
I see lightning CI failed. To reduce behaviour change, you can check it's ServerMode

Reply:
It was fixed, thanks! PTAL~

Reply:
For non-server mode it's a compatibility change. I need to ask PM about it tomorrow. It may break user's script.

Reply:
anything affected by current behavior of lightning? any workaround?

Reply:
In cloud, when k8s' job controller wants to schedule lightning pod to other node, job controller will send `SIGINT` to lightning pod. Because lightning pod returns 0, job controller think the job is done and will not schedule it.

Reply:
I added compatibility with the previous non-server mode. /cc @lance6716